modules/oe_authentication_user_fields/src/EuLoginAttributesHelper.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_authentication_user_fields;
+
+/**
+ * Helper functions to map EU Login attributes to user profile.
+ */
+class EuLoginAttributesHelper {
+
+  /**
+   * Array mapping EU Login attributes with user account fields.
+   */
+  const USER_EU_LOGIN_ATTRIBUTE_MAPPING = [
+    'mail' => 'email',
+    'field_oe_firstname' => 'firstName',
+    'field_oe_lastname' => 'lastName',
+    'field_oe_department' => 'departmentNumber',
+    'field_oe_organisation' => 'domain',
+  ];
+
+  /**
+   * Converts an array EU Login attributes into an array of Drupal field/values.
+   *
+   * @param array $attributes
+   *   An array containing a series of EU Login attributes.
+   *
+   * @return array
+   *   An array containing a series of Drupal field names and values.
```

Comment:
Isn't it more usual in Drupal to describe this as something like "An associative array of field values indexed by the field name"?